src/ansari/ansari_db.py

```diff
@@ -109,13 +112,95 @@ def _get_token_from_request(self, request: Request) -> str:
+            query (Union[str, List[str]]): A single SQL query string or a list of SQL query strings.
+            params (Union[tuple, List[tuple]]): A single tuple of parameters or a list of tuples of parameters.
+            which_fetch (Union[Literal["one", "all"], List[Literal["one", "all"]]]):
+                A single fetch type or a list of fetch types. Each fetch type can be:
+                - "one": Fetch one row.
+                - "all": Fetch all rows.
+                - Any other value: Do not fetch any rows.
+            commit_after (Literal["each", "all"]): Whether to commit the transaction after each query is executed,
+                or only after all of them are executed.
+
+        Returns:
+            Union[Optional[List], List[Optional[List]]]:
+                - If a single query is executed:
+                    - Returns a single result if which_fetch is "one"
+                    - Returns a single list of rows if which_fetch is "all".
+                    - Else, returns None.
+                - If multiple queries are executed:
+                    - Returns a list of results, where each result is:
+                        - A single result if which_fetch is "one".
+                        - A list of rows if which_fetch is "all".
+                        - Else, returns None.
+
+            Note: the "single result" here could be a tuple if more than 1 column is selected in the query.
+
+        Raises:
+            ValueError: If an invalid fetch type is provided.
+        """
+        # If query is a single string, we assume that params and which_fetch are also non-list values
+        if isinstance(query, str):
+            requested_single_query = True
+            query = [query]
+            params = [params]
+            which_fetch = [which_fetch]
+        # else, we assume that params and which_fetch are lists of the same length
+        # and do a list-conversion just in case they are strings
+        else:
+            requested_single_query = False
+            if isinstance(params, str):
+                params = [params] * len(query)
+            if isinstance(which_fetch, str):
+                which_fetch = [which_fetch] * len(query)
+
+        caller_function_name = inspect.stack()[1].function
+        logger.debug(f"Function {caller_function_name}() \nis running queries: \n{query} \nwith params: \n{params}")
+
+        results = []
+        with self.get_connection() as conn:
+            with conn.cursor() as cur:
+                for q, p, wf in zip(query, params, which_fetch):
+                    cur.execute(q, p)
+                    result = None
+                    if wf.lower() == "one":
+                        result = cur.fetchone()
+                    elif wf.lower() == "all":
+                        result = cur.fetchall()
+
+                    # Remove possible SQL comments at the start of the q variable
+                    q = re.sub(r"^\s*--.*\n", "", q, flags=re.MULTILINE)
```

Comment:
This feels a bit weird to me. Why is this necessary? Why is it two dashes instead of the normal 3? Does pysql not handle comments?

Reply:
I can adjust the regex to activate on 2 or 3 ISA ... actually, this regex isn't _that_ necessary 😅 , as we don't have any SQL statements starting with dashes in our code, but I will adjust the regex nonetheless 🙌